### PR TITLE
Small doc touchups and doctests for mv

### PR DESCRIPTION
**Pull request by kshyatt**
This passed doctests locally. All the changes are to the filesystem part of the docs (`@__LINE__` is displayed there).

### CODE REVIEW

**Comment by kshyatt** [comment]
I wanted to show all the behavior mentioned in the docstring. I can strip out the parts showing the file content hasn't changed if you'd like.

File: base/file.jl [hunk 1]
```diff
@@ -358,6 +358,43 @@ end
...
+ [1] #checkfor_mv_cp_cptree#10(::Bool, ::Function, ::String, ::String, ::String) at ./file.jl:293
+[...]
+
+julia> mv("hello.txt", "goodbye.txt", force=true)
+"goodbye.txt"
+
+julia> readline("goodbye.txt")
+"world2"
+
+julia> rm("goodbye.txt")
```
Comment (KristofferC) [inline base/file.jl]: Can be removed?

File: base/file.jl [hunk 2]
```diff
@@ -358,6 +358,43 @@ end
 Move the file, link, or directory from `src` to `dst`.
 `force=true` will first remove an existing `dst`.
 Return `dst`.
+
+# Examples
+```jldoctest; filter = r"Stacktrace:(\\n \\[[0-9]+\\].*)*"
+julia> write("hello.txt", "world");
+
+julia> "hello.txt" in readdir()
```
Comment (KristofferC) [inline base/file.jl]: Don't think we need to show that `write` creates the file here.

File: base/file.jl [hunk 3]
```diff
@@ -358,6 +358,43 @@ end
...
+julia> mv("hello.txt", "goodbye.txt")
+ERROR: ArgumentError: 'goodbye.txt' exists. `force=true` is required to remove 'goodbye.txt' before moving.
+Stacktrace:
+ [1] #checkfor_mv_cp_cptree#10(::Bool, ::Function, ::String, ::String, ::String) at ./file.jl:293
+[...]
+
+julia> mv("hello.txt", "goodbye.txt", force=true)
+"goodbye.txt"
+
+julia> readline("goodbye.txt")
```
Comment (KristofferC) [inline base/file.jl]: Don't need to show again that `mv` copies content again?

File: base/file.jl [hunk 4]
```diff
@@ -358,6 +358,43 @@ end
...
+julia> "hello.txt" in readdir()
+true
+
+julia> mv("hello.txt", "goodbye.txt")
+"goodbye.txt"
+
+julia> "hello.txt" in readdir()
+false
+
+julia> "goodbye.txt" in readdir()
```
Comment (KristofferC) [inline base/file.jl]: Can remove this since the line below shows that the file exist?

**Comment by kshyatt** [comment]
@KristofferC thanks for the review! Better now?

**Review by KristofferC — APPROVED** [review]

**Pull request merged**